zaza/openstack/charm_tests/ovn/tests.py

```diff
@@ -139,6 +140,97 @@ def setUpClass(cls):
                 'ovs-vswitchd',
             ]
 
+    def test_conf_enable_hardware_offload(self):
+        """Enable hardware offload and confirm unit configuration.
+
+        Note, this is a shallow non-blackbox test to confirm charm operation
+        only.
+        """
+        if self.current_release <= openstack_utils.get_os_release(
+                'bionic_ussuri'):
+            logging.info('bionic kernel does not have netdevsim, skip')
+            return
+        if self.application_name == 'ovn-dedicated-chassis':
+            logging.info('dedicated-chassis does curretly not support '
+                         'hardware offload, skip')
+            return
+        if zaza.utilities.machine_os.is_container(self.lead_unit):
+            logging.info('unit is in a container, skip')
+            return
+
+        ifnames = zaza.utilities.machine_os.add_netdevsim(
+            self.lead_unit, 10, 1)
```

Comment:
I'm guessing this permanently changes the unit under test? Should this be a in a configure step, or am I being to picky about it??

Reply:
You are right, I guess we could go even further and add clean-up steps to avoid permanently changing the unit too.